securesystemslib/spx_keys.py

```diff
@@ -0,0 +1,307 @@
+  """
+
+  # Generate spx's seed key by calling os.urandom().  The random bytes
+  # returned should be suitable for cryptographic use and is OS-specific.
+  # Raise 'NotImplementedError' if a randomness source is not found.
+  seed = os.urandom(pyspx.crypto_sign_SEEDBYTES)
+  public = None
+
+  # Generate the public key.  pyspx performs the actual key generation.
+  try:
+    public, private = pyspx.generate_keypair(seed)
+
+  except NameError: # pragma: no cover
+    raise securesystemslib.exceptions.UnsupportedLibraryError('The pyspx'
+        ' library and/or its dependencies unavailable.')
+
+  return public, private
+
+
+
+
+
+def create_signature(public_key, private_key, data, scheme):
+  """
+  <Purpose>
+    Return a (signature, scheme) tuple, where the signature scheme is 'spx'
+    and is always generated by pyspx.  The signature returned
+    conforms to 'securesystemslib.formats.SPXSIGNATURE_SCHEMA'.
+
+    >>> public, private = generate_public_and_private()
+    >>> data = b'The quick brown fox jumps over the lazy dog'
+    >>> scheme = 'spx'
+    >>> signature, scheme = \
+        create_signature(public, private, data, scheme)
+    >>> securesystemslib.formats.SPXSIGNATURE_SCHEMA.matches(signature)
+    True
+    >>> scheme == 'spx'
+    True
+    >>> signature, scheme = \
+        create_signature(public, private, data, scheme)
+    >>> securesystemslib.formats.SPXSIGNATURE_SCHEMA.matches(signature)
+    True
+    >>> scheme == 'spx'
+    True
+
+  <Arguments>
+    public:
+      The spx public key, a simple byte string
+
+    private:
+      The spx private key, a simple byte string
+
+    data:
+      Data object used by create_signature() to generate the signature.
+
+    scheme:
+      The signature scheme used to generate the signature.
+
+  <Exceptions>
+    securesystemslib.exceptions.FormatError, if the arguments are improperly
+    formatted.
+
+    securesystemslib.exceptions.CryptoError, if a signature cannot be created.
+
+  <Side Effects>
+    spx.signing.SigningKey.sign() called to generate the actual signature.
+
+  <Returns>
+    A signature dictionary conformat to 'securesystemslib.format.SIGNATURE_SCHEMA'.  
+  """
+
+  # Does 'public_key' have the correct format?
+  # This check will ensure 'public_key' conforms to
+  # 'securesystemslib.formats.SPXPUBLIC_SCHEMA', which must have length 32
+  # bytes.  Raise 'securesystemslib.exceptions.FormatError' if the check fails.
+  securesystemslib.formats.SPXPUBLIC_SCHEMA.check_match(public_key)
+
+  # Is 'private_key' properly formatted?
+  securesystemslib.formats.SPXSEED_SCHEMA.check_match(private_key)
+
+  # Is 'scheme' properly formatted?
+  securesystemslib.formats.SPX_SIG_SCHEMA.check_match(scheme)
+
+  # Signing the 'data' object requires a seed and public key.
+  # spx.signing.SigningKey.sign() generates the signature.
+  public = public_key
+  private = private_key
+
+  signature = None
+
+  # An if-clause is not strictly needed here, since 'spx' is the only
```

Comment:
Since we are bothering to handle the error with an `else:` clause, wouldn't you agree that the `if` check actually does matter? Maybe the comment could be reworded to reflect this if so.